### ISSUE TITLE
Inconsistent result after apply of datadog_synthetic_test resource

### ISSUE THREAD

**Issue by camlow325** [DataDog/terraform-provider-datadog#1302]
### Terraform Version

```
Terraform v1.0.8
+ provider registry.terraform.io/datadog/datadog v3.2.0
```

### Affected Resource(s)

- datadog_synthetic_test

### Terraform Configuration Files

I don't think the problem is specific to the content in the `datadog_synthetic_test` resources that we use but could provide a more specific example if that would be helpful.

### Debug Output

From the debug output, we see that the Datadog provider was able to create a synthetic test via the Datadog API:

```
2021-12-10T04:42:09.657Z [INFO]  provider.terraform-provider-datadog_v3.2.0: 2021/12/10 04:42:09
POST /api/v1/synthetics/tests/api HTTP/1.1
Host: api.datadoghq.com
...
HTTP/1.1 200 OK
...
{"status":"paused","public_id":"<id>",...
```

Immediately after that, the Datadog provider attempts to retrieve the test via the previously returned id. The Datadog service, however, returns an HTTP 404 error for this request:

```
GET /api/v1/synthetics/tests/<id> HTTP/1.1
Host: api.datadoghq.com
...
HTTP/1.1 404 Not Found
...
{"errors": ["Synthetics test not found"]}
```

This, then, appears to lead to the resource erroring out:

```
2021-12-10T04:42:10.233Z [TRACE] maybeTainted: datadog_synthetics_test.this encountered an error during creation, so it is now marked as tainted
2021-12-10T04:42:10.233Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState to workingState for datadog_synthetics_test.this
2021-12-10T04:42:10.233Z [TRACE] NodeAbstractResouceInstance.writeResourceInstanceState: removing state object for datadog_synthetics_test.this
2021-12-10T04:42:10.233Z [TRACE] evalApplyProvisioners: datadog_synthetics_test.this is tainted, so skipping provisioning
2021-12-10T04:42:10.233Z [TRACE] maybeTainted: datadog_synthetics_test.this was already tainted, so nothing to do
```

### Expected Behavior

Apply should succeed.

### Actual Behavior

Apply generally succeeds, but we occasionally see errors like the following:

```
Error: Provider produced inconsistent result after apply

When applying changes to datadog_synthetics_test.this,
provider "provider[\"registry.terraform.io/datadog/datadog\"]" produced an
unexpected new value: Root resource was present, but now absent.
```

When the error occurs, it does appear that a Terraform apply will succeed on a second try, so this is what we're doing to work around the problem for now.

It wasn't immediately clear whether there is already some retry logic in the provider, like around the read code [here](https://github.com/DataDog/terraform-provider-datadog/blob/v3.6.0/datadog/resource_datadog_synthetics_test_.go#L857), for cases where the API returns an HTTP 404. It may be useful to add this if such logic does not already exist and there are eventual consistency considerations around resource creation with the Datadog API. The AWS provider has code like this for a number of resources.

### Steps to Reproduce

`terraform apply`

**Comment by therve** [comment]
Hi, I'm pretty sure this is fixed by #1117 , which is in version 3.3.0. Can you try upgrading? Thanks.

**Comment by camlow325** [comment]
Ah, thanks. That definitely seems possible. I'll go ahead and close this for now and we'll upgrade to the new version. Thanks much for the response and insight!
